railties/lib/rails/generators/rails/app/templates/config/puma.rb

```diff
@@ -0,0 +1,9 @@
+threads_count = ENV.fetch("MAX_THREADS") { 5 }.to_i
```

Comment:
I like to be able to change my server behavior without having to wait for an entire code deploy. I see this line as a benefit for those who can configure env separately from code and nothing lost for those who only change configurations via code/deploys. 

Not now, but eventually if we could standardize on a environment variable we could use it to set all locations that have a thread pool, so changing in one place would work in all locations. For example imagine if we had this code and also had a database.yml

```yml
production:
  pool: <%= ENV.fetch("MAX_THREADS") { 5 } %>
```

Now when you increase your puma threads, your AR pool is increased automatically and it does the right thing out of the box. Right now you have to know EVERY place that uses a thread pool when you change your puma config. Sidekiq, AR, sucker_punch, AMQP or other stores, threaded API clients. If you change one but not all of them, then you get errors like `ActiveRecord::ConnectionTimeoutError`. Even if you're not configuring using env vars, it gives a standard thing to grep for to change all those values. Having all of those components work together by a convention is the direction I would eventually like to see default Rails config, but again that's an "eventually" goal, not within the scope of this PR.